al_aws_collector.js

```diff
@@ -148,23 +152,45 @@ class AlAwsCollector {
             });
     }
     
-    checkin(event, context, callback) {
-        // TODO: add stats, etc
+    checkin(callback) {
         var collector = this;
-        var checks = this._customHealthChecks;
-        m_healthChecks.getHealthStatus(event, context, checks, function(status) {
-            const checkinValues = Object.assign(collector._getAttrs(), status);
-            collector._azcollectc.checkin(checkinValues)
-            .then(resp => {
-                return callback(null);
-            })
-            .catch(exception => {
-                return callback(exception);
-            });
+        const context = this._invokeContext;
+        const checks = this._customHealthChecks;
+        const statsFuns = this._customStatsFuns;
+
+        async.parallel([
+            function(asyncCallback) {
+                m_healthChecks.getHealthStatus(context, checks, function(err, healthStatus) {
+                    return asyncCallback(err, healthStatus);
+                });
+            },
+            function(asyncCallback) {
+                m_stats.getStatistics(context, statsFuns, function(err, response) {
+                    return asyncCallback(err, {statistics : response});
```

Comment:
I think we should not fail checkin if we aren't able to get stats.
Let's always return `asyncCallback(null, {statistics : response});` and put some error message in to response. Or just return empty stats.

Reply:
Or I guess stats error will be there already https://github.com/alertlogic/al-aws-collector-js/pull/10/files#diff-6744f4713d4b5329a930e46bd8c1c4cbR31

Reply:
@kkuzmin yup, we actually never return error : 
https://github.com/alertlogic/al-aws-collector-js/blob/c7bca9d16a7b4b2972f68dece15f6282fb23e7e8/statistics_templates.js#L62-L72

```
        if (err) {
            return callback(null, {
                Label: params.MetricName,
                StatisticsError: JSON.stringify(err).slice(0, MAX_ERROR_MSG_LEN)
            });
        } else {
            return callback(null, {
                Label: data.Label,
                Datapoints: data.Datapoints
            });
        }
```

however this line might be confusing, and it's better to have more bullet-proof way of not returning errors